### PR TITLE
introduce LcmService

### LINKED ISSUE #586: Introduce support for lcmEntities
Add support for `/gdc/account/profile/.../lcmEntities`

### PR DESCRIPTION
**Pull request by jimirocks**
Resolves #586

### CODE REVIEW

File: src/main/java/com/gooddata/lcm/LcmEntitiesDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import com.gooddata.collections.PageableListDeserializer;
+import com.gooddata.collections.Paging;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Deserializer of {@link LcmEntities}.
+ */
+class LcmEntitiesDeserializer extends PageableListDeserializer<LcmEntities, LcmEntity> {
```
Comment (liry) [inline src/main/java/com/gooddata/lcm/LcmEntitiesDeserializer.java]: What about to add these (de)serializers as a static inner classes of LcmEntities?

File: src/main/java/com/gooddata/lcm/LcmEntityFilter.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        params = new LinkedMultiValueMap<>(3);
+    }
+
+    /**
+     * Adds given data product to this filter.
+     * @param dataProduct data product id - must not be empty.
+     * @return this filter
+     */
+    public LcmEntityFilter withDataProduct(final String dataProduct) {
+        params.put("dataProduct", singletonList(notEmpty(dataProduct, "dataProduct")));
```
Comment (liry) [inline src/main/java/com/gooddata/lcm/LcmEntityFilter.java]: What about to use constants for the key strings?

File: src/main/java/com/gooddata/lcm/LcmEntitiesDeserializer.java (hunk 1 above)
Comment (jimirocks) [reply src/main/java/com/gooddata/lcm/LcmEntitiesDeserializer.java]: done

File: src/main/java/com/gooddata/lcm/LcmEntityFilter.java (hunk 2 above)
Comment (jimirocks) [reply src/main/java/com/gooddata/lcm/LcmEntityFilter.java]: fixed

File: src/main/java/com/gooddata/lcm/LcmEntityFilter.java [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+public class LcmEntityFilter {
+
+    private static final String DATA_PRODUCT = "dataProduct";
+    private static final String SEGMENT = "segment";
+    private static final String CLIENT = "client";
+
+    private String dataProduct;
+    private String segment;
+    private String client;
+    private final MultiValueMap<String, String> params;
```
Comment (liry) [inline src/main/java/com/gooddata/lcm/LcmEntityFilter.java]: plz remove this, it's not used anymore

Comment (jimirocks) [reply src/main/java/com/gooddata/lcm/LcmEntityFilter.java]: done

File: src/main/java/com/gooddata/lcm/LcmService.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+     */
+    public LcmService(final RestTemplate restTemplate, final GoodDataSettings settings) {
+        super(restTemplate, settings);
+    }
+
+    /**
+     * Lists all {@link  LcmEntities} for given {@link Account}.
+     * Returns empty list in case there is no {@link LcmEntity}.
+     * Returns only first page if there's more instances than page limit. Use {@link PageableList#stream()} to iterate
+     * over all pages, or {@link MultiPageList#collectAll()} to load the entire list.
```
Comment (liry) [inline src/main/java/com/gooddata/lcm/LcmService.java]: @jimirocks @jakub-plichta why are we guiding user to use MultiPageList, but returning only PageableList? Is the MultiPageList meant to be used directly by end user?

**Comment by liry** [comment]
the rest LGTM

**Pull request merged**